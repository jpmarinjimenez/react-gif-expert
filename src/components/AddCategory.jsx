import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
    const minLenght = 4;
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();

        const newInputValue = inputValue.trim();

        if (newInputValue.length < minLenght) return;

        onNewCategory(newInputValue);
        setInputValue('');
    };

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input type="text" placeholder="Buscar Gif" value={inputValue} onChange={onInputChange} autoFocus />
        </form>
    );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
};
