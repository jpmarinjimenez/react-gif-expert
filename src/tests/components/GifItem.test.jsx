import { render } from '@testing-library/react';
import { GifItem } from '../../components/GifItem';

describe('Testing GifItem component', () => {
    const title = 'Testing GifItem';
    const url =
        'https://media2.giphy.com/media/TjAcxImn74uoDYVxFl/giphy-downsized-medium.gif?cid=96343e7asjat8zr3wcx7x1vcdj1u8iiann5m9mpsgav7m0rm&rid=giphy-downsized-medium.gif&ct=g';

    test('should match the snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });
});
