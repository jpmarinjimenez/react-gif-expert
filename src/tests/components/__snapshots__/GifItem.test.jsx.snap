// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Testing GifItem"
      src="https://media2.giphy.com/media/TjAcxImn74uoDYVxFl/giphy-downsized-medium.gif?cid=96343e7asjat8zr3wcx7x1vcdj1u8iiann5m9mpsgav7m0rm&rid=giphy-downsized-medium.gif&ct=g"
    />
    <p>
      Testing GifItem
    </p>
  </div>
</div>
`;
